OpenRA.Mods.Common/Traits/Player/ProductionQueue.cs

```diff
@@ -468,11 +481,37 @@ bool CancelProductionInner(string itemName)
 		public void EndProduction(ProductionItem item)
 		{
 			Queue.Remove(item);
+
+			if (item.Infinite)
+			{
+				item.Reset();
```

Comment:
I'd prefer if we could avoid defining this `Reset` method.  Is there a reason why you can't do
```csharp
Queue.Add(new ProductionItem(this, item.Item, item.TotalCost, playerPower, item.OnComplete));
```
like above?